examples/bench/main.go

```diff
@@ -0,0 +1,1020 @@
+			ops[1].ColorM.Scale(0.2, 1.0, 0.2, 1.0)
+			for i := 0; i < b.N; i++ {
+				idx := i
+				ops[idx%2].GeoM.Reset()
+				ops[idx%2].GeoM.Translate(float64(i*320/b.N), float64(i*200/b.N))
+				screen.DrawImage(px26, ops[idx%2])
+			}
+		},
+	},
+	{
+		name: "draw26color10",
+		fn: func(b *testing.B, screen *ebiten.Image) {
+			ops := []*ebiten.DrawImageOptions{&ebiten.DrawImageOptions{}, &ebiten.DrawImageOptions{}}
+			ops[0].ColorM.Scale(1.0, 0.2, 0.2, 1.0)
+			ops[1].ColorM.Scale(0.2, 1.0, 0.2, 1.0)
+			for i := 0; i < b.N; i++ {
+				idx := i / 10
+				ops[idx%2].GeoM.Reset()
+				ops[idx%2].GeoM.Translate(float64(i*320/b.N), float64(i*200/b.N))
+				screen.DrawImage(px26, ops[idx%2])
+			}
+		},
+	},
+}
+
+var (
+	benchIdx  = 0
+	results   []testing.BenchmarkResult
+	didSetup  sync.Once
+	benchFunc func(b *testing.B, screen *ebiten.Image)
+	benchGo   = make(chan struct{})
+	benchDone = make(chan bool)
+)
+
+func benchmarks(screen *ebiten.Image) error {
+	var setupErr error
+
+	didSetup.Do(func() {
+		eimg, _, err := openEbitenImage()
+		if err != nil {
+			setupErr = fmt.Errorf("can't open image: %s", err)
+			return
+		}
+		px26 = eimg
+
+		w, h := eimg.Size()
+		img, err := ebiten.NewImage(w*4, h*4, ebiten.FilterNearest)
+		if err != nil {
+			setupErr = fmt.Errorf("can't create ne image: %s", err)
+			return
+		}
+		px104 = img
+
+		op := &ebiten.DrawImageOptions{}
+		op.GeoM.Scale(4, 4)
+		px104.DrawImage(px26, op)
+
+		img, err = ebiten.NewImage(w*16, h*16, ebiten.FilterNearest)
+		if err != nil {
+			setupErr = fmt.Errorf("can't create ne image: %s", err)
+			return
+		}
+		px416 = img
+		op.GeoM.Scale(4, 4)
+		px416.DrawImage(px26, op)
+	})
+	if setupErr != nil {
+		return setupErr
+	}
+
+	if benchFunc != nil {
+		benchGo <- struct{}{}
+		if <-benchDone {
+			benchFunc = nil
+		}
+	} else {
+		if len(benchList) > benchIdx {
+			benchFunc = benchList[benchIdx].fn
+			go func(idx int) {
+				benchList[idx].result = testing.Benchmark(func(b *testing.B) {
+					b.StopTimer()
+					ebiten.SetBenchmark(b)
+					var elapsed time.Duration
+					// run for at least 60 frames, so the FPS value should mean something
+					for i := 0; i < 60; i++ {
+						<-benchGo
+						t0 := time.Now()
+						b.StartTimer()
+						benchFunc(b, screen)
+						b.StopTimer()
+						t1 := time.Now()
+						elapsed += t1.Sub(t0)
+						benchDone <- false
+					}
+					fmt.Printf("%16s: N %5d, FPS: %5.2f, time %10v/60 frames\n", benchList[idx].name, b.N, ebiten.CurrentFPS(), elapsed)
+					ebiten.SetBenchmark(nil)
+				})
+				<-benchGo
+				benchDone <- true
+			}(benchIdx)
+			benchIdx++
+		} else {
+			return noErr
+		}
+	}
+	return nil
+}
+
+func openEbitenImage() (*ebiten.Image, image.Image, error) {
+	img, _, err := image.Decode(bytes.NewReader(images.Ebiten_png))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	eimg, err := ebiten.NewImageFromImage(img, ebiten.FilterNearest)
+	if err != nil {
+		return nil, nil, err
+	}
+	return eimg, img, nil
+}
+
```

Comment:
I think you copied this from `image_test.go`, and you don't need the below code.

Reply:
ayup. i was going to add more things, then forgot.

it's a pretty awful hack, it's mostly there to try to serve as a proof of concept.